### PR TITLE
SASS-2051 - Add pensions summary page and initial plumbing

### PR DESCRIPTION
**Pull request by HarryGull**
### Description
Pensions - Create pensions summary page and add plumbing and test data builders

[SASS-2051](https://jira.tools.tax.service.gov.uk/browse/SASS-2051)

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/controllers/pensions/PensionsSummaryController.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import services.PensionSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.pensions.PensionsSummaryView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PensionsSummaryController @Inject()(
+                                           appConfig: AppConfig,
```
Comment (sraspin) [inline app/controllers/pensions/PensionsSummaryController.scala:33]: could make this 'implicit val appConfig: AppConfig' and then remove it from line 40

File: app/controllers/pensions/PensionsSummaryController.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                                           implicit val ec: ExecutionContext,
+                                           authAction: AuthorisedAction,
+                                           pensionSessionService: PensionSessionService,
+                                           pensionSummaryView: PensionsSummaryView) extends FrontendController(mcc) with I18nSupport with Logging {
+
+  implicit val config: AppConfig = appConfig
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+
+    pensionSessionService.getAndHandle(taxYear, redirectWhenNoPrior = false) { (_, prior) =>
```
Comment (sraspin) [inline app/controllers/pensions/PensionsSummaryController.scala:44]: 'redirectWhenNoPrior = false' is the default case so you could remove it

File: it/controllers/pensions/PensionsSummaryControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    val updated = "Updated"
+    val cannotUpdate = "Cannot update"
+    val toDo = "To do"
+  }
+
+  object ExpectedIndividualEN extends SpecificExpectedResults {
+    val expectedH1 = "Pensions"
+    val expectedTitle = "Pensions"
+
+    def expectedContent(taxYear: Int): String = s"You cannot change your employment information until 6 April $taxYear."
```
Comment (sraspin) [inline it/controllers/pensions/PensionsSummaryControllerISpec.scala:86]: do we need this?

File: it/controllers/pensions/PensionsSummaryControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+  object ExpectedIndividualCY extends SpecificExpectedResults {
+    val expectedH1: String = "Pensions"
+    val expectedTitle: String = "Pensions"
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    val expectedH1: String = "Pensions"
+    val expectedTitle: String = "Pensions"
+
+    def expectedContent(taxYear: Int): String = s"You cannot change your client’s employment information until 6 April $taxYear."
```
Comment (sraspin) [inline it/controllers/pensions/PensionsSummaryControllerISpec.scala:103]: Same here, is this needed?

File: it/controllers/pensions/PensionsSummaryControllerISpec.scala (hunk 3 above)
Comment (HarryGull) [reply it/controllers/pensions/PensionsSummaryControllerISpec.scala]: Removed

File: it/controllers/pensions/PensionsSummaryControllerISpec.scala (hunk 4 above)
Comment (HarryGull) [reply it/controllers/pensions/PensionsSummaryControllerISpec.scala]: removed

**Review by sraspin — APPROVED** [review]

**Pull request merged**